.github/workflows/test-internal-prs.yml

```diff
@@ -0,0 +1,169 @@
+# Description: This workflow runs test suite against PRs that are not from forks.
+#
+# Triggered by:
+#   (1) Opening a pull request
+#   (2) On every commit to the PR
+#   (3) Adding run-tests label to the PR
+
+name: Test / Internal PRs
+
+concurrency:
+  group: test-internal-prs-${{ github.event.pull_request.id }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
```

Comment:
This used to be `pull_request_target`. Changing this to `pull_request` is a big security win because `pull_request` will not pass any secrets or write-level GH token to forks.